OWML.ModHelper.Menus/LayoutManager.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Reflection;
+using UnityEngine;
+using UnityEngine.UI;
+using OWML.Common.Menus;
+using System.Collections.Generic;
+using OWML.ModHelper.Events;
+
+namespace OWML.ModHelper.Menus
+{
+    public class LayoutManager : ILayoutManager
+    {
+        private const int FontSize = 36;
+        private static readonly Vector2 NormalPivot = new Vector2(0.5f, 0.5f);
+
+        public LayoutGroup LayoutGroup { get; private set; }
+        public int ChildCount => LayoutGroup.transform.childCount;
+
+        private readonly UIStyleManager _styleManager;
+        private readonly ModUIStyleApplier _styleApplier;
+        private readonly Vector3 _scale;
+        private readonly HashSet<Graphic> _constantGraphics = new HashSet<Graphic>();
+        private readonly HashSet<Graphic> _backingGraphics = new HashSet<Graphic>();
+
+        public LayoutManager(LayoutGroup layout, UIStyleManager styleManager, ModUIStyleApplier styleApplier, Vector3 scale, Graphic[] constantGraphics, Graphic[] backGraphics) : this(layout, styleManager, styleApplier, scale, constantGraphics)
+        {
+            styleApplier.SetBackround(backGraphics);
+        }
+
+        public LayoutManager(LayoutGroup layout, UIStyleManager styleManager, ModUIStyleApplier styleApplier, Vector3 scale, Graphic[] constantGraphics) : this(layout, styleManager, styleApplier, scale)
+        {
+            Array.ForEach(constantGraphics, element => _constantGraphics.Add(element));
+        }
+
+        public LayoutManager(LayoutGroup layout, UIStyleManager styleManager, ModUIStyleApplier styleApplier, Vector3 scale)
+        {
+            _scale = scale;
+            _styleManager = styleManager;
+            _styleApplier = styleApplier;
+            LayoutGroup = layout;
+            LayoutGroup.childAlignment = TextAnchor.MiddleCenter;
+            LayoutGroup.transform.localPosition = Vector3.zero;
+            ((RectTransform)LayoutGroup.transform).pivot = NormalPivot;
+            UpdateState();
+        }
+
+        public void UpdateState()
+        {
+            var currentTexts = LayoutGroup.gameObject.GetComponentsInChildren<Text>();
+            _styleApplier.SetTexts(currentTexts);
+            Graphic[] currentGraphics = currentTexts;
+            Array.Resize<Graphic>(ref currentGraphics, currentTexts.Length + _constantGraphics.Count);
+            int i = currentTexts.Length;
+            foreach (var graphic in _constantGraphics)
+            {
+                currentGraphics[i] = graphic;
+                i++;
+            }
+            _styleApplier.SetForeground(currentGraphics);
+        }
+
+        public void Clear()
+        {
+            var childCount = LayoutGroup.transform.childCount;
+            for (var i = childCount - 1; i >= 0; i--)
```

Comment:
foreach(Transform child in LayoutGroup.transform)

Reply:
K